### PR TITLE
feat: allow filtering activities by type

### PR DESCRIPTION
**Pull request by sborrazas**

### CODE REVIEW

File: test/ae_mdw_web/controllers/activities_controller_test.exs [hunk 1]
```diff
@@ -1023,180 +1023,265 @@ defmodule AeMdwWeb.ActivitiesControllerTest do
...
+        )
+        |> Store.put(
+          Model.AexnTransfer,
+          Model.aexn_transfer(index: {:aex141, next_account_pk, 5, account_pk, 2, 2})
+        )
+        |> Store.put(Model.Tx, Model.tx(index: 5, block_index: {height, mbi}, id: "tx-hash5"))
+        |> Store.put(
+          # Skipped
+          Model.TargetKindIntTransferTx,
+          Model.target_kind_int_transfer_tx(index: {account_pk, "reward_oracle", {height, 6}, 6})
```
Comment (jyeshe) [inline test/ae_mdw_web/controllers/activities_controller_test.exs:1113]: Made sense for me to skip due to the passive role while receiving the reward, even when the oracle executes the query itself. Likewise might be the case to add the `oracle_query_tx` with the `oracle_id` position [here](https://github.com/aeternity/ae_mdw/blob/6631dee6a7449240e8c3b59f3aabedd68719e26a/lib/ae_mdw/fields.ex#L21) as the oracle receives the request to execute the query.

**Review by jyeshe — COMMENTED** [review]
LGTM for the filtering by type. Reviewing the test cases got some particular cases on other parts of the code that affect activities.

File: lib/ae_mdw/activities.ex [hunk 2]
```diff
@@ -688,5 +754,12 @@ defmodule AeMdw.Activities do
 
   defp convert_param({"owned_only", _val}), do: {:ok, {:ownership_only?, true}}
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply lib/ae_mdw/activities.ex]: I've now added the same handling for other boolean params

File: test/ae_mdw_web/controllers/activities_controller_test.exs [hunk 3]
```diff
@@ -1023,180 +1023,265 @@ defmodule AeMdwWeb.ActivitiesControllerTest do
...
+                 "type" => "ContractCallTxEvent",
+                 "payload" => %{"micro_index" => ^mbi}
+               } = tx2
+
+        assert %{
+                 "height" => ^height,
+                 "block_hash" => ^enc_mb_hash,
+                 "type" => "Aex9TransferEvent",
+                 "payload" => %{"amount" => 1}
+               } = tx3
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply test/ae_mdw_web/controllers/activities_controller_test.exs:1154]: Done

File: test/ae_mdw_web/controllers/activities_controller_test.exs (hunk 1 above)
Comment (sborrazas) [reply test/ae_mdw_web/controllers/activities_controller_test.exs:1113]: Added it as an excluded pair, thanks

**Review by jyeshe — CHANGES_REQUESTED** [review]
With `sender_id: account_id` the transaction should be returned with `owned_only`. Chaging the test case to a different spender is not fixing the problem.

**Review by jyeshe — APPROVED** [review]

**Pull request merged**